app/decorators/physical_server_decorator.rb

```diff
@@ -0,0 +1,9 @@
+class PhysicalServerDecorator < MiqDecorator
+  def fileicon
+    "svg/vendor-#{vendor.try(:downcase)}.svg"
```

Comment:
Well, if it happens only when you manually delete it or where the factory is wrongly configured, I think we can ignore those cases.. :)

(Fixing the factory should be as simple as adding a `:vendor => 'foo'` when calling `FactoryGirl.build`)